[#43182] Migrate permissions of existing admins and users to handle Nextcloud operations

fraber opened the pull request:
https://community.openproject.org/work_packages/43182

Implemented preliminary rules to update permissions for Storages.
Somebody with these permissions -> will get these new permissions:
- `view_work_packages` --> `view_file_links`
- `select_project_modules` --> `manage_storages_in_project`
- `edit_work_packages` --> `manage_file_links`

fraber committed:
Implemented preliminary rules to update permissions for Storages

Somebody with these permissions -> will get these new permissions:
- view_work_packages -> view_file_links
- select_project_modules -> manage_project_storages
- edit_work_packages -> manage_file_links

wielinde reviewed (commented):
@fraber Cool, that you found that migration helper. Code looks good to me. I didn't have the time to test it manually.

@apfohl Could you please also review the code and do a manual test as described in the ticket? Thank you.

apfohl commented:
> @apfohl Could you please also review the code and do a manual test as described in the ticket? Thank you.

Sure.

wielinde commented on modules/storages/db/migrate/20220712165928_add_storage_file_link_perms.rb:
```diff
@@ -0,0 +1,49 @@
+           :view_file_links)
+
+    ::Migration::MigrationUtils::PermissionAdder
+      .add(:edit_work_package,
```
```suggestion
      .add(:edit_work_packages,
```

wielinde commented on modules/storages/db/migrate/20220712165928_add_storage_file_link_perms.rb:
```diff
@@ -0,0 +1,49 @@
+  def up
+    ::Migration::MigrationUtils::PermissionAdder
+      .add(:select_project_modules,
+           :manage_project_storages)
```
```suggestion
           :manage_storages_in_project)
```

wielinde committed:
Fix permission names

wielinde committed:
Update and rename 20220712165928_add_storage_file_link_perms.rb to 20220712165928_add_storages_permissions_to_roles.rb

I believe it is better not to introduce abbreviations unless it is not necessary. Also the migration is not strictly about file links.
